pkg/provider/aws/resources/cloudfront.go

```diff
@@ -82,56 +83,26 @@ type (
 	}
 )
 
-// CreateS3Origin creates an origin for a static unit, given its bucket, and attaches it to the Cloudfront distribution passed in
-func CreateS3Origin(unit *core.StaticUnit, bucket *S3Bucket, distribution *CloudfrontDistribution, dag *core.ResourceGraph) {
+type OriginAccessIdentityCreateParams struct {
+	Name string
+	Refs core.AnnotationKeySet
+}
 
-	oai := &OriginAccessIdentity{
-		Name:          fmt.Sprintf("%s-%s", bucket.Name, unit.ID),
-		ConstructsRef: core.AnnotationKeySetOf(unit.AnnotationKey),
-		Comment:       "this is needed to setup s3 polices and make s3 not public.",
-	}
-
-	policyDoc := &PolicyDocument{
-		Version: VERSION,
-		Statement: []StatementEntry{
-			{
-				Effect: "Allow",
-				Principal: &Principal{
-					AWS: core.IaCValue{
-						Resource: oai,
-						Property: IAM_ARN_IAC_VALUE,
-					},
-				},
-				Action: []string{"s3:GetObject"},
-				Resource: []core.IaCValue{
-					{
-						Resource: bucket,
-						Property: ALL_BUCKET_DIRECTORY_IAC_VALUE,
-					},
-				},
-			},
-		},
-	}
-	bucketPolicy := NewBucketPolicy(unit.ID, bucket, policyDoc)
-	dag.AddDependency(bucketPolicy, oai)
-	dag.AddDependency(distribution, oai)
-	dag.AddDependency(bucketPolicy, bucket)
-	s3OriginConfig := S3OriginConfig{
-		OriginAccessIdentity: core.IaCValue{
-			Resource: oai,
-			Property: CLOUDFRONT_ACCESS_IDENTITY_PATH_IAC_VALUE,
-		},
+func (oai *OriginAccessIdentity) Create(dag *core.ResourceGraph, params OriginAccessIdentityCreateParams) error {
+	oai.Name = params.Name
+	oai.ConstructsRef = params.Refs
+	if dag.GetResource(oai.Id()) != nil {
+		return fmt.Errorf(`Origin Access Identity with name "%s" already exists`, oai.Name)
```

Comment:
do we want to fail here? What if someone had 2 cdns with the same bucket? I know we dont support that now but just a what if

Reply:
the oai is bucket level so it should be reuseable across cdns if needed, thus not needing a new id. We dont support a use case that needs it yet so not blocking